src/nvim/option.c

```diff
@@ -4667,7 +4664,9 @@ char_u *check_colorcolumn(win_T *wp)
       col = col * getdigits_int(&s);
       if (wp->w_buffer->b_p_tw == 0)
         goto skip;          /* 'textwidth' not set, skip this item */
-      col += wp->w_buffer->b_p_tw;
+      assert(wp->w_buffer->b_p_tw >= (long)INT_MIN - col
+             && wp->w_buffer->b_p_tw <= (long)INT_MAX - col);
```

Comment:
Arrrgh. Right :+1: Former expression would be valid if all terms were int. But not when we're checking for narrowing type conversion. Your precision continues to marvel me, @oni-link ;-)